extension/parquet/parquet_crypto.cpp

```diff
@@ -358,15 +358,30 @@ uint32_t ParquetCrypto::WriteData(TProtocol &oprot, const const_data_ptr_t buffe
 	return etrans.Finalize();
 }
 
+std::string Base64Decode(const std::string &key) {
+	auto result_size = Blob::FromBase64Size(key);
+	auto output = duckdb::unique_ptr<unsigned char[]>(new unsigned char[result_size]);
+	Blob::FromBase64(key, output.get(), result_size);
+	std::string decoded_key(reinterpret_cast<const char *>(output.get()), result_size);
+	return decoded_key;
+}
+
 void ParquetCrypto::AddKey(ClientContext &context, const FunctionParameters &parameters) {
 	const auto &key_name = StringValue::Get(parameters.values[0]);
 	const auto &key = StringValue::Get(parameters.values[1]);
-	if (!AESGCMState::ValidKey(key)) {
-		throw InvalidInputException(
-		    "Invalid AES key. Must have a length of 128, 192, or 256 bits (16, 24, or 32 bytes)");
-	}
+
 	auto &keys = ParquetKeys::Get(context);
-	keys.AddKey(key_name, key);
+	if (AESGCMState::ValidKey(key)) {
+		keys.AddKey(key_name, key);
+	} else {
+		// try Base64 decoding
+		std::string decoded_key = Base64Decode(key);
```

Comment:
Try catch around Base64Decode to always return:
```
throw InvalidInputException(
			    "Invalid AES key. Not a plain AES key NOR a base64 encoded string")
```
?

I think it's ConversionError is rather weird otherwise.

---

I see this being useful, unsure if it might not worth to have directly the CREATE SECRET parquet (either as part of parquet or directly in 1.0.0 via secondary parquet extension).